components/formats-common/test/loci/common/utests/DateToolsTest.java

```diff
@@ -306,10 +310,14 @@ public void testGetTimeConflictingSeparator() {
   @Test
   public void testUnrecognizedTrailingPattern() {
     // Date parsing should fail but the method should still return
-    String date = REFERENCE_DATE + ".111 wrong";
-    assertEquals(-1, DateTools.getTime(date, DATE_FORMAT, "."));
-    assertEquals(null, DateTools.formatDate(date, DATE_FORMAT, "."));
-    assertEquals(null, DateTools.formatDate(date, DATE_FORMAT, false, "."));
+    String date1 = REFERENCE_DATE + ".000 AM";  // space separated
+    assertEquals(-1, DateTools.getTime(date1, DATE_FORMAT, "."));
+    assertEquals(null, DateTools.formatDate(date1, DATE_FORMAT, "."));
+    assertEquals(null, DateTools.formatDate(date1, DATE_FORMAT, false, "."));
+    String date2 = REFERENCE_DATE + ".000Z";  // non-digit character
+    assertEquals(-1, DateTools.getTime(date2, DATE_FORMAT, "."));
+    assertEquals(null, DateTools.formatDate(date2, DATE_FORMAT, "."));
+    assertEquals(null, DateTools.formatDate(date2, DATE_FORMAT, false, "."));
```

Comment:
There is an `assertNull` but it suffers from the same limitations as `assertTrue/assertFalse` (it is using `assertTrue` internally) in the sense that the error thrown in case of test failure will not return the expected vs real value.